### ISSUE TITLE
Feature Request: Detect unreachable Target URLs

### ISSUE THREAD

**Issue by raamdev** [raamdev/wp-redirects#28]
It would be nice if WP Redirects could attempt to reach Target URLs when saving a redirect and notify the site owner when a 404 or some other error is returned.

It might also be nice to have an option that lets WP Redirects occasionally poll each Target URL to confirm the URL is still reachable and if it becomes unreachable to notify the site owner in some way, either via a dismissable Dashboard message, by a graphical indicator in the redirect row when viewing all redirects.

It would also be useful to have another column called "Target Status" that reports either "OK" or "Error"; that way a site owner with _many_ redirects could easily sort the list by all redirects that have an error.

**Comment by jaswrks** [comment]
:+1: Love that idea. It would be really helpful. I'm about to bring a _lot_ of Redirects from an old site into a new site. When I do, having the ability to check the status of each would be awesome.

**Comment by raamdev** [comment]
> I'm about to bring a lot of Redirects from an old site into a new site. When I do, having the ability to check the status of each would be awesome.

@jaswsinc Ah, good to know! I'll make this a higher priority in that case. Thanks for the heads up! Any idea (even a rough one) when you'll be doing that, so that I can better plan when I need to have this feature added by?

**Comment by jaswrks** [comment]
Approx. March 4th :-)

**Comment by raamdev** [comment]
@jaswsinc Thank you. I'm setting the WP-Redirects Next Release target date for Feb 27th, to include this feature.

**Comment by jaswrks** [comment]
It might be cool (perhaps later) to also add an automatic screen grab to this feature. So in addition to checking the status code, we could provide a thumbnail of the source URL also.

See: http://www.websharks-inc.com/thumbshot/generator.php
~ click link at the top to reveal details about our API.

**Why?** Sometimes the status will be as expected, but the underlying resource might have changed. Scanning through a list of statuses, together with thumbnails, might make it easier to spot problems.

I'm imagining a config. option that would allow you to set a URL that could be used to configure the thumbnail API you want to use. In our case, we could use our own utility for this (URL above). However, it might default to one of many free services available for this.

**Comment by raamdev** [comment]
> It might be cool (perhaps later) to also add an automatic screen grab to this feature. So in addition to checking the status code, we could provide a thumbnail of the source URL also.

Interesting idea! I assume such thumbnails would need to be cached/stored somewhere, right? To avoid having to re-fetch them unnecessarily? Or would you want them to always be re-fetched (when they're not cached in the browser) so as to ensure you're seeing the most up-to-date thumbnail?

**Comment by jaswrks** [comment]
> I assume such thumbnails would need to be cached/stored somewhere, right?

Yes. Most of the APIs will cache the images for you automatically. However, depending on the way you integrate it, it could require a cache be built by WP Redirects. Some of the APIs use a simple REST approach, while others will come with a PHP SDK and deliver raw binary data.

In either case, I think a cache of 7 days or less would be fine. Ideally, a cache time for this specific use case might be around 48 hours. The [tool that I built](http://www.websharks-inc.com/thumbshot/generator.php) has a default cache expiration time of 30 days. We'd probably want to change that for this use case.
